x/protorev/types/errors.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+import (
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var (
+	ErrInvalidRoute       = sdkerrors.Register(ModuleName, 1, "This is not a valid cyclic route")
+	ErrDuplicateTokenPair = sdkerrors.Register(ModuleName, 2, "This token pair has already been added")
+	ErrInvalidTokenName   = sdkerrors.Register(ModuleName, 3, "This is not a valid token name")
+	ErrInvalidParams      = sdkerrors.Register(ModuleName, 4, "Invalid params")
+	ErrInvalidArbDenom    = sdkerrors.Register(ModuleName, 5, "Invalid denom for arbitrage")
```

Comment:
I recommend not doing registered errors, since the codespaces make error type not safely changeable in patch releases. Perhaps just an errors.New or fmt.ErrorF() ?

Reply:
Got it. We will refactor and replace the registered errors with fmt.ErrorF().